### PR TITLE
Fix reference axes menu item

### PR DESCRIPTION
**Pull request by ghutchis**
Signed-off-by: Geoff Hutchison <geoff.hutchison@gmail.com>

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ghutchis** [comment]
Not sure why the axes no longer show up though.. I'll leave this to @serk12

**Comment by ghutchis** [comment]
The problem is that tools and scene plugins `emit drawablesChanged();` to update the render.
Right now, there's no way to do that in `extensionsplugin.h` @serk12 

This should be discussed with @cryos - perhaps it's better to revert to the display type from an extension.

**Review by serk12 — COMMENTED** [review]
I see two issues. 
First, on opening `setActiveWidget` it's not called and `processAxis` does nothing because `m_glWidget` it's null. 

Second, once the user does an action (navigation, add atoms, etc... `m_glWidget` is no longer null) the button gets de-sync with the axes, this code fix this.
```
--- a/avogadro/qtplugins/overlayaxes/overlayaxes.cpp
+++ b/avogadro/qtplugins/overlayaxes/overlayaxes.cpp
@@ -239,9 +239,9 @@ OverlayAxes::OverlayAxes(QObject* parent_)
   connect(m_axesAction, SIGNAL(triggered()), SLOT(processAxis()));
 
   QSettings settings;
-  m_enabled = settings.value("overlayAxes/enabled", true).toBool();
+  m_enabled = !settings.value("overlayAxes/enabled", true).toBool();
   m_axesAction->setCheckable(true);
-  m_axesAction->setChecked(m_enabled);
+  processAxis();
 }
 
 OverlayAxes::~OverlayAxes()
@@ -274,7 +274,9 @@ void OverlayAxes::processAxis()
     m_scene->rootNode().removeChild(engineNode);
     delete engineNode;
     m_widgetToNode[m_glWidget] = nullptr;
-  } else {
+  }
+
+  if (m_enabled) {
     engineNode = new Rendering::GroupNode(&node);
     m_widgetToNode[m_glWidget] = engineNode;
     process(*m_molecule, *engineNode);
```

File: avogadro/qtplugins/overlayaxes/overlayaxes.cpp [hunk 1]
```diff
@@ -274,7 +276,9 @@ void OverlayAxes::processAxis()
     m_scene->rootNode().removeChild(engineNode);
     delete engineNode;
     m_widgetToNode[m_glWidget] = nullptr;
-  } else {
+  }
+
+  if (m_enabled) {
     engineNode = new Rendering::GroupNode(&node);
```
Comment (ghutchis) [inline avogadro/qtplugins/overlayaxes/overlayaxes.cpp:282]: @serk12 - on launch, the code crashes here...

Comment (serk12) [reply avogadro/qtplugins/overlayaxes/overlayaxes.cpp:282]: There is a memory leak with `m_scene` when the `OverlayAxes` constructor calls `processAxes` remove it  (line 246?) and it should work.

Comment (ghutchis) [reply avogadro/qtplugins/overlayaxes/overlayaxes.cpp:282]: @serk12 - that removes the crash, but then the user doesn't see the axes when enabled. Maybe running `processAxes` when `setMolecule` is called?

I'm wondering if it isn't worth reverting to a `render` plugin.

**Comment by ghutchis** [comment]
This one finishes it - state is remembered and the axes appear in the scene "on launch" (i.e., when `m_scene` is set).

**Pull request merged**